pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -308,6 +313,10 @@ protected BrokerResponse handleRequest(long requestId, String query, SqlNodeAndO
         }
       }
 
+      if (isDefaultQueryResponseLimitEnabled() && !pinotQuery.isSetLimit()) {
```

Comment:
Will we ever hit this? In `PinotQuery` thrift definition, limit has a default value 10. Will limit always be set?